tests/python/test_tools_smoke.py

```diff
@@ -0,0 +1,371 @@
+#!/usr/bin/env python
+# Copyright (c) Sasha Goldshtein, 2017
+# Licensed under the Apache License, Version 2.0 (the "License")
+
+import distutils.version
+import subprocess
+import os
+from unittest import main, skipUnless, TestCase
+
+TOOLS_DIR = "../../tools/"
+
+def kernel_version_ge(major, minor):
+    # True if running kernel is >= X.Y
+    version = distutils.version.LooseVersion(os.uname()[2]).version
+    if version[0] > major:
+        return True
+    if version[0] < major:
+        return False
+    if minor and version[1] < minor:
+        return False
+    return True
+
+@skipUnless(kernel_version_ge(4,1), "requires kernel >= 4.1")
+class SmokeTests(TestCase):
+    # Use this for commands that have a built-in timeout, so they only need
+    # to be killed in case of a hard hang.
+    def run_with_duration(self, command, timeout=10):
+        full_command = TOOLS_DIR + command
+        self.assertEqual(0,     # clean exit
+                subprocess.call("timeout -s KILL %ds %s > /dev/null" %
+                                (timeout, full_command), shell=True))
+
+    # Use this for commands that don't have a built-in timeout, so we have
+    # to Ctrl-C out of them by sending SIGINT. If that still doesn't stop
+    # them, send a kill signal 5 seconds later.
+    def run_with_int(self, command, timeout=5, allow_early=False, kill=False):
+        full_command = TOOLS_DIR + command
+        signal = "KILL" if kill else "INT"
+        rc = subprocess.call("timeout -s %s -k %ds %ds %s > /dev/null" %
+                             (signal, 5, timeout, full_command), shell=True)
```

Comment:
1 second is not enough because a lot of tools take a while to initialize, especially on a slow system. Sending a SIGINT while they are still initializing sort of defeats the purpose of the smoke test.